### PR TITLE
Adding `Dry::Container::Mixin#each_pair`

### PR DESCRIPTION
**Pull request by jeremyf**
In discussions with other developers, it was felt that being able to
iterate over not just the registered keys, but to see what was
registered would be very helpful. This is a step toward doing that.

### CODE REVIEW

**Comment by jeremyf** [comment]
Adding issue concerning rbx-2 (#27)

**Comment by AMHOL** [comment]
Thanks for the PR.

We have discussed this in the past and decided against adding it as nobody had a specific need for it, could you explain your use-case please?

File: lib/dry/container/mixin.rb [hunk 1]
```diff
@@ -179,6 +179,22 @@ def each_key(&block)
...
+      # @return [Dry::Container::Mixin] self
+      #
+      # @api public
+      #
+      # @note In discussions with other developers, it was felt that being able to iterate over not just
+      #       the registered keys, but to see what was registered would be very helpful. This is a step
+      #       toward doing that.
+      def each_pair(&block)
+        config.resolver.each_pair(_container, &block)
+        self
```
Comment (AMHOL) [inline lib/dry/container/mixin.rb:195]: Still not sure on the feature as a whole yet, but I think that returning `self` here is a bad idea, it doesn't seem like a good candidate for method chaining and encourages mutation in some form or another

**Comment by solnic** [comment]
Why `each_pair` instead of just `each` yielding a key and an object?

**Comment by AMHOL** [comment]
@jeremyf going to close this issue due to inactivity, please respond if this is still an issue that you think needs addressing and we can re-open

**Comment by jeremyf** [comment]
@AMHOL I switched to `#each` with a new pull request #30

**Pull request closed without merging**